packages/stores/src/account/osmosis/index.ts

```diff
@@ -1205,6 +1206,80 @@ export class OsmosisAccountImpl {
     );
   }
 
+  /**
+   *
+   * @param routes Routes to split swap through.
+   * @param tokenIn Token swapping in.
+   * @param tokenOutMinAmount Minimum amount of token out expected.
+   * @param numTicksCrossed Number of CL ticks crossed for swap quote.
+   * @param memo Transaction memo.
+   * @param TxFee Fee options.
+   * @param signOptions Signing options.
+   * @param onFulfill Callback to handle tx fullfillment given raw response.
+   */
+  async sendSplitRouteSwapExactAmountOutMsg(
+    routes: {
+      pools: {
+        id: string;
+        tokenInDenom: string;
+      }[];
+      tokenOutAmount: string;
+    }[],
+    tokenOut: { coinMinimalDenom: string },
+    tokenInMaxAmount: string,
+    memo: string = "",
+    signOptions?: SignOptions & { fee?: StdFee },
+    onFulfill?: (tx: DeliverTxResponse) => void
+  ) {
+    const msg = await makeSplitRoutesSwapExactAmountOutMsg({
+      routes,
+      tokenOut,
+      tokenInMaxAmount,
+      userOsmoAddress: this.address,
+    });
+
+    await this.base.signAndBroadcast(
+      this.chainId,
+      "splitRouteSwapExactAmountOut",
+      [msg],
+      memo,
+      signOptions?.fee,
+      signOptions,
+      (tx) => {
+        if (!tx.code) {
+          // Refresh the balances
+          const queries = this.queriesStore.get(this.chainId);
+          queries.queryBalances
+            .getQueryBech32Address(this.address)
+            .balances.forEach((_bal) => {
+              //TODO: DOES THIS NEED REFRESHED?
+              // if (
+              //   bal.currency.coinMinimalDenom ===
+              //     tokenIn.currency.coinMinimalDenom ||
+              //   routes
+              //     .flatMap(({ pools }) => pools)
+              //     .find(
+              //       (pool) =>
+              //         pool.tokenOutDenom === bal.currency.coinMinimalDenom
+              //     )
+              // ) {
+              //   bal.waitFreshResponse();
+              // }
```

Comment:
If we want to delete this, we need to determine in this PR whether it's necessary. We should test it with the pool details page. Otherwise, let's leave it uncommented